src/main/scala/is/hail/expr/ir/Infer.scala

```diff
@@ -7,9 +7,10 @@ object Infer {
     ir match {
       case If(cond, cnsq, altr) =>
         assert(cond.typ.isOfType(TBoolean()))
-        assert(cnsq.typ == altr.typ, s"mismatch:\n  ${ cnsq.typ.parsableString() }\n  ${ altr.typ.parsableString() }\n  $cond")
-        cnsq.typ
-
+        if (cnsq.typ != altr.typ)
```

Comment:
can you keep an assert here? `cnsq.typ isOfType altr.typ` or something

Reply:
yes, that seems like an absolute requirement. good call.